### DISCUSSION TITLE
Windows installation

### DISCUSSION THREAD

**Discussion by nagordon** [DLR-AE/LoadsKernel#50] [Questions & Answers] [answered]
Just wanted to share that the windows installation is a little tricky. This worked for me with the loadskernel and model-viewer

```
conda create --name loadskernel conda 
conda activate loadskernel 
pip install mayavi
pip install PyQt5
pip install configobj
git clone https://github.com/DLR-AE/LoadsKernel.git
cd LoadsKernel
pip install -e .[extras]
```

**Comment by ArneVoss** [comment]
Thanks for sharing! I usually use the software on my Linux workstation.

Have you tried the conda package as an alternative? On my Windows laptop that package worked fine with Python 3.10 and 3.11.

**Reply by nagordon** [reply]
yes I just tested and that works too. I was having issues with my older conda install so I used pip, but would recommend using conda as you indicated. 

UPDATE: the conda package is old and has some bugs that are fixed in the newer version. I think the reason the pip install worked is because I cloned the latest repo and installed locally. Can you update the conda package to the latest version? Thanks.

**Reply by ArneVoss** [reply]
I just wanted to let you know there is a new Release 2025.01 available, including new conda packages.

**Comment by entropia666** [comment]
Hi! I just saw this thread and want to share our experience on the installation (of the 2024 release), in particular the vtk package gave us headaches, (we had [this error](https://github.com/conda-forge/vtk-feedstock/issues/207) on both windows and macOS)we managed to install on windows and MacOS:
```
`conda config --add channels conda-forge`
`conda config --set channel_priority strict`
`conda create -n py_dlr_loads python=3.11.10`
`conda activate py_dlr_loads`
`conda install vtk` #not sure this is needed
`conda install loadskernel`
`conda install mayavi`
`conda install vtk -c conda-forge --force-reinstall`
```

I also managed to install it in development mode, after the above steps:

```
conda activate py_dlr_loads
cd C:\Users\Michele\LoadsKernel
pip install -e .
```

Thanks for your work!

**Reply by ArneVoss** [reply, marked as answer]
While the conda package 'loadskernel' includes all dependencies to run the core software, there is a conda package 'loadskernel-extras' that should include all dependencies (like mayavi) for the graphical tools. I am fully aware that the graphical packages can cause some trouble, which is the reason for the separation into two packages.

conda install -c conda-forge loadskernel-extras

**Reply by entropia666** [reply]
Thanks for your reply! I just tested it works, I don't know why we didn't see it in November when we installed it...
